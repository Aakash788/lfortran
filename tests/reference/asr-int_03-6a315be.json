{
    "basename": "asr-int_03-6a315be",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/int_03.f90",
    "infile_hash": "cff22b3361cf7567f02e0efffa4556d2d81e76a78438598aead45a25",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-int_03-6a315be.stdout",
    "stdout_hash": "339aa278e2b1d5a43201b12655dac0ff06d1bcb53d46525203616171",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}