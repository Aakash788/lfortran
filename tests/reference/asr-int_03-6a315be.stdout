(TranslationUnit 
    (SymbolTable
        1
        {
            int_03:
                (Program 
                    (SymbolTable
                        2
                        {
                            ans:
                                (Variable 
                                    2 
                                    ans 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Integer 8 [((IntegerConstant 1 (Integer 4 [])) 
                                    (IntegerConstant 5 (Integer 4 [])))]) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            i:
                                (Variable 
                                    2 
                                    i 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            w:
                                (Variable 
                                    2 
                                    w 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Complex 8 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    int_03 
                    [] 
                    [(= 
                        (ArrayItem 
                            (Var 2 ans) 
                            [(() 
                            (IntegerConstant 1 (Integer 4 [])) 
                            ())] 
                            (Integer 8 []) 
                            ()
                        ) 
                        (IntegerConstant 8524933037632333570 (Integer 8 [])) 
                        ()
                    )
                    (= 
                        (ArrayItem 
                            (Var 2 ans) 
                            [(() 
                            (IntegerConstant 2 (Integer 4 [])) 
                            ())] 
                            (Integer 8 []) 
                            ()
                        ) 
                        (IntegerUnaryMinus 
                            (IntegerConstant 1069250973542918798 (Integer 8 [])) 
                            (Integer 8 []) 
                            (IntegerConstant -1069250973542918798 (Integer 8 []))
                        ) 
                        ()
                    )
                    (= 
                        (ArrayItem 
                            (Var 2 ans) 
                            [(() 
                            (IntegerConstant 3 (Integer 4 [])) 
                            ())] 
                            (Integer 8 []) 
                            ()
                        ) 
                        (IntegerUnaryMinus 
                            (IntegerConstant 5123867065024149335 (Integer 8 [])) 
                            (Integer 8 []) 
                            (IntegerConstant -5123867065024149335 (Integer 8 []))
                        ) 
                        ()
                    )
                    (= 
                        (ArrayItem 
                            (Var 2 ans) 
                            [(() 
                            (IntegerConstant 4 (Integer 4 [])) 
                            ())] 
                            (Integer 8 []) 
                            ()
                        ) 
                        (IntegerConstant 7303655603304982073 (Integer 8 [])) 
                        ()
                    )
                    (= 
                        (ArrayItem 
                            (Var 2 ans) 
                            [(() 
                            (IntegerConstant 5 (Integer 4 [])) 
                            ())] 
                            (Integer 8 []) 
                            ()
                        ) 
                        (IntegerConstant 5108441843522503546 (Integer 8 [])) 
                        ()
                    )
                    (Print 
                        () 
                        [(Var 2 ans)] 
                        () 
                        ()
                    )
                    (= 
                        (Var 2 w) 
                        (Cast 
                            (ComplexConstructor 
                                (RealConstant 
                                    7.700000 
                                    (Real 4 [])
                                ) 
                                (RealConstant 
                                    5.000000 
                                    (Real 4 [])
                                ) 
                                (Complex 4 []) 
                                (ComplexConstant 
                                    7.700000 
                                    5.000000 
                                    (Complex 4 [])
                                )
                            ) 
                            ComplexToComplex 
                            (Complex 8 []) 
                            ()
                        ) 
                        ()
                    )
                    (= 
                        (Var 2 i) 
                        (Cast 
                            (Var 2 w) 
                            ComplexToInteger 
                            (Integer 4 []) 
                            ()
                        ) 
                        ()
                    )
                    (Print 
                        () 
                        [(Var 2 i)] 
                        () 
                        ()
                    )
                    (If 
                        (IntegerCompare 
                            (Var 2 i) 
                            NotEq 
                            (IntegerConstant 7 (Integer 4 [])) 
                            (Logical 4 []) 
                            ()
                        ) 
                        [(ErrorStop 
                            ()
                        )] 
                        []
                    )]
                )
            
        }) 
    []
)
